tests/unittest/disaggregated/test_mamba_transfer.py

```
@@ -321,6 +322,111 @@ def test_mamba_policy_slot_major_interleaved_role_ptrs():
+            receiver_ri=rank_info,
+        )
+        == 72
+    )
+
+
+def test_mamba_policy_rejects_incompatible_recurrent_side_state():
+    self_group = _make_mamba_group(base_address=1000)
+    peer_group = _make_mamba_group(base_address=10000, side_bytes=16)
+    rank_info = _make_rank_info()
+
+    with pytest.raises(ValueError, match="side-state role.*slot_bytes differs"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)